### PR TITLE
chore: use lambda.Runtime.NODEJS_12_X in all tests

### PR DESCRIPTION
**Pull request by trivikr**
Follow-up to https://github.com/aws/aws-cdk/pull/12713

This PR uses lambda.Runtime.NODEJS_12_X in CDK tests.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ses/lib/receipt-rule.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export class DropSpamReceiptRule extends CoreConstruct {
     super(scope, id);
 
     const fn = new lambda.SingletonFunction(this, 'Function', {
-      runtime: lambda.Runtime.NODEJS_10_X,
+      runtime: lambda.Runtime.NODEJS_12_X,
```
Comment (nija-at) [inline packages/@aws-cdk/aws-ses/lib/receipt-rule.ts:178]: This is not changing the test but actually changing the underlying custom resource. When customers upgrade, they're going to have their CF custom resource replaced. We'll need to make sure that the underlying handler is Node12 compatible.

Can we pull this change out and create a separate PR for these changes, for deeper scrutiny and customer notification?

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts [hunk 2]
```diff
@@ -58,7 +58,7 @@ export class EvaluateExpression extends sfn.TaskStateBase {
   constructor(scope: Construct, id: string, private readonly props: EvaluateExpressionProps) {
     super(scope, id, props);
 
-    this.evalFn = createEvalFn(this.props.runtime || lambda.Runtime.NODEJS_10_X, this);
+    this.evalFn = createEvalFn(this.props.runtime || lambda.Runtime.NODEJS_12_X, this);
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts:61]: same here.

File: packages/@aws-cdk/custom-resources/lib/provider-framework/provider.ts [hunk 3]
```diff
@@ -172,7 +172,7 @@ export class Provider extends CoreConstruct implements cfn.ICustomResourceProvid
     const fn = new lambda.Function(this, `framework-${entrypoint}`, {
       code: lambda.Code.fromAsset(RUNTIME_HANDLER_PATH),
       description: `AWS CDK resource provider framework - ${entrypoint} (${this.node.path})`.slice(0, 256),
-      runtime: lambda.Runtime.NODEJS_10_X,
+      runtime: lambda.Runtime.NODEJS_12_X,
```
Comment (nija-at) [inline packages/@aws-cdk/custom-resources/lib/provider-framework/provider.ts:175]: and here

**Review by nija-at — DISMISSED** [review]
Nice! Thanks for making these changes. Just one comment below.

File: packages/@aws-cdk/aws-ses/lib/receipt-rule.ts (hunk 1 above)
Comment (trivikr) [reply packages/@aws-cdk/aws-ses/lib/receipt-rule.ts]: Reverted and created an issue instead https://github.com/aws/aws-cdk/issues/13223

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts (hunk 2 above)
Comment (trivikr) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts]: Reverted and created a follow-up issue https://github.com/aws/aws-cdk/issues/13224

File: packages/@aws-cdk/custom-resources/lib/provider-framework/provider.ts (hunk 3 above)
Comment (trivikr) [reply packages/@aws-cdk/custom-resources/lib/provider-framework/provider.ts]: Reverted and created issue for follow-up https://github.com/aws/aws-cdk/issues/13225

**Comment by jogold** [comment]
Why not go directly for `NODEJS_14_X`?

**Comment by nija-at** [comment]
> Why not go directly for `NODEJS_14_X`?

I believe CloudFormation does not yet support inline code for Node 14.
As of 2 weeks ago, it seems this remains to be the case - https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/80#issuecomment-777194191

(I've not tested this personally)

**Review by nija-at — CHANGES_REQUESTED** [review]
Could you look at the build failures? Looks like there are more changes to be made in the tests.

You will also need to address the merge conflicts :(

Ping me when it's ready and I'll try to ship it before any further merge conflicts arise.

**Comment by trivikr** [comment]
Packaged custom resources were moved to nodejs 14.x in https://github.com/aws/aws-cdk/pull/13488
